iwyu_include_picker.h

```diff
@@ -67,6 +67,8 @@ struct IncludeMapEntry;
 
 enum class RegexDialect;
 enum IncludeVisibility { kUnusedVisibility, kPublic, kPrivate };
+enum LibCImpl { kNoC, kGlibc };
+enum LibCXXImpl { kNoCXX, kLibstdcpp };
```

Comment:
Clang calls this `CXXStdlibType`: https://github.com/llvm/llvm-project/blob/main/clang/include/clang/Driver/ToolChain.h#L95.

I think good names would be:
```
CStdlib
CXXStdlib
```
to indicate the relationship with the Clang construct (that said, ours are only used as "mapping keys", and we will probably need many more sets of mappings than Clang needs to care about individual standard libs, so they're not the same. Just closely related.)

I would also prefer to use `enum class` for any new enums, so how about:
```
enum class CStdLib {
  None,
  Glibc
};

enum class CXXStdLib {
  None,
  Libstdcxx
};
```